pkg/tokenvalidation/tokenauthenticator.go

```diff
@@ -0,0 +1,84 @@
+package oauth
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kauthenticator "k8s.io/apiserver/pkg/authentication/authenticator"
+	kuser "k8s.io/apiserver/pkg/authentication/user"
+
+	authorizationv1 "github.com/openshift/api/authorization/v1"
+	oauthclient "github.com/openshift/client-go/oauth/clientset/versioned/typed/oauth/v1"
+	userclient "github.com/openshift/client-go/user/clientset/versioned/typed/user/v1"
+)
+
+var errLookup = errors.New("token lookup failed")
+
+type tokenAuthenticator struct {
+	tokens       oauthclient.OAuthAccessTokenInterface
+	users        userclient.UserInterface
+	groupMapper  UserToGroupMapper
+	validators   OAuthTokenValidator
+	implicitAuds kauthenticator.Audiences
+}
+
+func NewTokenAuthenticator(tokens oauthclient.OAuthAccessTokenInterface, users userclient.UserInterface, groupMapper UserToGroupMapper, implicitAuds kauthenticator.Audiences, validators ...OAuthTokenValidator) kauthenticator.Token {
+	return &tokenAuthenticator{
+		tokens:       tokens,
+		users:        users,
+		groupMapper:  groupMapper,
+		validators:   OAuthTokenValidators(validators),
+		implicitAuds: implicitAuds,
+	}
+}
+
+func (a *tokenAuthenticator) AuthenticateToken(ctx context.Context, name string) (*kauthenticator.Response, bool, error) {
+	token, err := a.tokens.Get(context.TODO(), name, metav1.GetOptions{})
```

Comment:
an access token is set by `oauth-server` after successful login, correct ?

Reply:
the `oauth-server` creates it, yes